mongo/integration/sdam_prose_test.go

```diff
@@ -108,4 +109,37 @@ func TestSDAMProse(t *testing.T) {
 			assert.Soon(t, callback, defaultCallbackTimeout)
 		})
 	})
+
+	mt.RunOpts("client waits between failed Hellos", mtest.NewOptions().MinServerVersion("4.9"), func(mt *mtest.T) {
+		// Force hello requests to fail 5 times.
+		mt.SetFailPoint(mtest.FailPoint{
+			ConfigureFailPoint: "failCommand",
+			Mode: mtest.FailPointMode{
+				Times: 5,
+			},
+			Data: mtest.FailPointData{
+				FailCommands: []string{internal.LegacyHello, "hello"},
+				ErrorCode:    1234,
+				AppName:      "SDAMMinHeartbeatFrequencyTest",
+			},
+		})
+
+		// Create Client and Connect after setting failpoint and before Ping.
+		clientOpts := options.Client().SetDirect(true).
+			SetAppName("SDAMMinHeartbeatFrequencyTest").
+			SetServerSelectionTimeout(5 * time.Second)
+		client, err := mongo.NewClient(clientOpts)
```

Comment:
Will this fail for some CI environments where the server URI isn't `localhost:27017`?

Consider using `mt.ResetClient` instead. E.g.
```go
opts := options.Client().
	SetDirect(true).
	SetAppName("SDAMMinHeartbeatFrequencyTest").
	SetServerSelectionTimeout(5 * time.Second)
mt.ResetClient(opts)
mt.Client.Ping(...)
// ...
```

Reply:
Yep.. Didn't know about the `ResetClient` function, thanks! Updated.